src/diffusers/models/vq_model.py

```diff
@@ -129,11 +132,13 @@ def encode(self, x: torch.FloatTensor, return_dict: bool = True) -> VQEncoderOut
 
     @apply_forward_hook
     def decode(
-        self, h: torch.FloatTensor, force_not_quantize: bool = False, return_dict: bool = True
+        self, h: torch.FloatTensor, force_not_quantize: bool = False, return_dict: bool = True, shape=None
     ) -> Union[DecoderOutput, torch.FloatTensor]:
         # also go through quantization layer
         if not force_not_quantize:
             quant, _, _ = self.quantize(h)
+        elif h.dtype == torch.long:
+            quant = self.quantize.get_codebook_entry(h, shape)
```

Comment:
What I call get_codebook_entry manually and then pass the result into decode, the forward hook isn't applied to the quantize params stay on the cpu. I added the apply_forward_hook to `get_codebook_entry` but that didn't work. I'm ok with this because it lets us pass both tokens and embeddings to the decode method. I'm also ok with not making this change but need some help on the accelerate side in that case

Reply:
Would this work: https://github.com/huggingface/diffusers/pull/5437/files#r1400946034?

Reply:
resolving to use other discussion thread